libraries/AP_HAL_Linux/HAL_Linux_Class.cpp

```diff
@@ -361,6 +361,10 @@ void HAL_Linux::run(int argc, char* const argv[], Callbacks* callbacks) const
         CMDLINE_SERIAL9,
     };
 
+    // set explicit line buffering for e.g. logging via systemd
+    setvbuf(stdout, (char *)0, _IOLBF, 0);
+    setvbuf(stderr, (char *)0, _IOLBF, 0);
```

Comment:
NULL or nullptr could be used.

Reply:
It's just copied from elsewhere to keep things uniform:
1. https://github.com/ArduPilot/ardupilot/blob/f0900bd119562dffe4cafabc1d2903e6951c128f/libraries/AP_HAL_SITL/SITL_cmdline.cpp#L369
2. https://github.com/ArduPilot/ardupilot/blob/f0900bd119562dffe4cafabc1d2903e6951c128f/libraries/AP_HAL_SITL/SITL_Periph_State.cpp#L63

Should I change to `nullptr` in those instances as well? I also am a little unsure if we really need unbuffered output or maybe line buffering would suffice. It definitely would for HAL Linux case, but with SITL it's not super clear as it may be doing some printing without a line feed which I'm not aware of.